_posts/2017-01-30-docker-frontend-tools.md

```diff
@@ -0,0 +1,146 @@
+
+While we could install these dependencies locally, that defeats the purpose of using Docker. One of the great advantages of virtualization, be it Docker or a full VM, is that you don't have to rely on installing global dependencies on your local machine. One of the many benefits of this is that it ensures each team member is developing in the same environment.
+
+Unfortunately, while Docker4Drupal provides public images for many applications (such as Nginx, PHP, MariaDB, Mailhog, Redis, Apache Solr, and Varnish), it does not provide images for running the Pattern Lab Starter theme dependencies.
+
+One of the nice features of Docker though is that it is relatively easy to create a new image that builds upon other images. This is done via a `Dockerfile` which specifies the commands for creating the image.
+
+To build an image with our theme's dependencies we created a `Dockerfile` with the following contents:
+
+```text
+FROM node:7.1
+MAINTAINER Dan Murphy <dan@savaslabs.com>
+
+RUN apt-get update && \
+    apt-get install -y php5-dev  && \
+    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer && \
+
+    # Directory required by Yeoman to run.
+    mkdir -p /root/.config/configstore \
+
+    # Clean up.
+    apt-get clean && \
+    rm -rf \
+			/root/.composer \
+			/tmp/* \
+			/usr/include/php \
+			/usr/lib/php5/build \
+			/var/lib/apt/lists/*
+
+# Permissions required by Yeoman to run: https://github.com/keystonejs/keystone/issues/1566#issuecomment-217736880
+RUN chmod g+rwx /root /root/.config /root/.config/configstore
+
+EXPOSE 3001 3050
+```
+
+The commands in this `Dockerfile`:
+
+- Set [the official Node 7 image](https://hub.docker.com/_/node/) as the base image. This base image includes Node and npm.
+- Install PHP 5 and Composer
+- Make configuration changes necessary for running Yeoman
+- Expose ports 3001 and 3050 which are necessary for serving the Pattern Lab style guide.
+
+From this `Dockerfile` we built the image `savaslabs/node-php-composer` and made it [publicly available on DockerHub](https://hub.docker.com/r/savaslabs/node-php-composer/).
+
+One piece of advice I have for building images for local development is that while Alpine Linux based images may be much smaller in size, the bare-bones nature and lack of common packages makes it much more difficult to build upon. For that reason, we based our image on the standard `DebianJessie` Node image rather than the Alpine variant.
+
+Now a Docker purist might critique this image and recommend only “one process per container”. However, since this image is for local development, isn't being used to deploy a production app, and encapsulates all of the applications required by the Pattern Lab Starter theme, we felt comfortable with this approach.
+
+### Using the image
+
+To use this image, we specified it in our project's `docker-compose.yml` file by adding the following lines to the services section:
```

Comment:
@dmurphy1 could you add a (sanitized) version of the complete `docker-compose.yml` file to some directory in this repo ( @AnneTee any ideas where would be the best place to store that?) and link to that so that folks can view the full file/